From bc227c756caafe5533582e5539b96f9d3a2f1da3 Mon Sep 17 00:00:00 2001
From: luca020400 <luca.stefani.ge1@gmail.com>
Date: Fri, 7 Nov 2014 18:10:42 +0100
Subject: [PATCH] wifi: add support for custom Wi-Fi implementations

---
 wifi/Android.mk | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/wifi/Android.mk b/wifi/Android.mk
index 99a4b0e..028480e 100644
--- a/wifi/Android.mk
+++ b/wifi/Android.mk
@@ -25,7 +25,13 @@ ifdef WIFI_DRIVER_FW_PATH_PARAM
 LOCAL_CFLAGS += -DWIFI_DRIVER_FW_PATH_PARAM=\"$(WIFI_DRIVER_FW_PATH_PARAM)\"
 endif
 
-LOCAL_SRC_FILES += wifi/wifi.c
+ifeq ($(TARGET_CUSTOM_WIFI),)
+       LOCAL_SRC_FILES += \
+               wifi/wifi.c
+else
+       LOCAL_SRC_FILES += \
+               $(TARGET_CUSTOM_WIFI)
+endif
 
 ifdef WPA_SUPPLICANT_VERSION
 LOCAL_CFLAGS += -DLIBWPA_CLIENT_EXISTS
-- 
1.9.1

